frontend/src/components/common/SideBar.tsx

```diff
@@ -1,30 +1,114 @@
 import React from "react";
-import { Box, Image } from "@chakra-ui/react";
-import RHSLogo from "../../assets/RHSlogo.png";
-
-const SideBar = ({
-  children,
-}: {
-  children?: React.ReactNode;
-}): React.ReactElement => {
+import {
+  Box,
+  Flex,
+  Text,
+  Spacer,
+  Image,
+  Button
+} from "@chakra-ui/react";
+
+
+import CoursesIcon from "../../assets/Courses.svg";
+import UsersIcon from "../../assets/Users.svg";
+import RHSLogo from "../../assets/RHSLogo-white.png";
+
+
+
+const Sidebar = (): React.ReactElement => {
   return (
-    <Box
-      className="sidebar"
-      width="280px"
-      minWidth="280px"
-      backgroundColor="brand.royal"
-    >
-      <Image
-        src={RHSLogo}
-        alt="Rowan House logo"
-        width="150px"
-        mx={16}
-        my={8}
-      />
-      {children}
-      <p>Test element</p>
+    <Box w="20%">
+      <Flex
+        position="fixed"
+        flex = "1"
+        top="0"
+        left="0"
+        p="4px"
+        w="280px"
+        h="100%"
+        minW="min-content"
+        boxShadow="xl"
+        flexFlow="column"
+        flexDir="column"
+        alignItems="center"
+        bg = "brand.royal"
+        color ="white"
+        orientation = "vertical"
+        justify="space=between"
+      >
+      
+      <Flex>
+        <Image
```

Comment:
Same as the above, not sure why there's so many newlines and so much space between JSX elements. It makes the code hard to read.